src/parser.js

```diff
@@ -0,0 +1,782 @@
+              this.makeAnnotation(ANNOTATIONS.VALIDATION_ERROR, item.path,
+                item.message);
+
+              if (item.inner) {
+                // TODO: I am honestly not sure what the correct behavior is
+                // here. Some items will have within them a tree of other items,
+                // some of which might contain more info (but it's unclear).
+                // Do we treat them as their own error or do something else?
+                queue.push(item.inner);
+              }
+            }
+            queue.shift();
+          }
+        }
+      }
+
+      const api = new Category();
+      this.api = api;
+      parseResult.push(api);
+
+      // Root API Element
+      api.classes.push('api');
+
+      if (swagger.info) {
+        this.path.push('info');
+
+        if (swagger.info.title) {
+          api.meta.set('title', swagger.info.title);
+
+          if (this.generateSourceMap) {
+            this.makeSourceMap(api.meta.get('title'), this.path.concat(['title']));
+          }
+        }
+
+        if (swagger.info.description) {
+          api.content.push(new Copy(swagger.info.description));
+
+          if (this.generateSourceMap) {
+            this.makeSourceMap(api.content[api.content.length - 1],
+              this.path.concat(['description']));
+          }
+        }
+
+        this.path.pop();
+      }
+
+      if (swagger.host) {
+        let hostname = swagger.host;
+
+        if (swagger.schemes) {
+          if (swagger.schemes.length > 1) {
+            this.makeAnnotation(ANNOTATIONS.DATA_LOST, ['schemes'],
+              'Only the first scheme will be used to create a hostname');
+          }
+
+          hostname = `${swagger.schemes[0]}://${hostname}`;
+        }
+
+        api.attributes.set('meta', {});
+        const meta = api.attributes.get('meta');
+        const member = new MemberElement('HOST', hostname);
+        member.meta.set('classes', ['user']);
+
+        if (this.generateSourceMap) {
+          this.makeSourceMap(member, ['host']);
+        }
+
+        meta.content.push(member);
+      }
+
+      if (swagger.securityDefinitions) {
+        this.makeAnnotation(ANNOTATIONS.DATA_LOST, ['securityDefinitions'],
+          'Authentication information is not yet supported');
+      }
+
+      if (swagger.security) {
+        this.makeAnnotation(ANNOTATIONS.DATA_LOST, ['security'],
+          'Authentication information is not yet supported');
+      }
+
+      if (swagger.externalDocs) {
+        this.makeAnnotation(ANNOTATIONS.DATA_LOST, ['externalDocs'],
+          'External documentation is not yet supported');
+      }
+
+      this.group = api;
+
+      // Swagger has a paths object to loop through
+      // The key is the href
+      _.each(_.omit(swagger.paths, isExtension), this.handlePath.bind(this));
+
+      done(null, parseResult);
+    });
+  }
+
+  // == Internal properties & functions ==
+
+  handlePath(pathValue, href) {
```

Comment:
I was a tad confused of the naming of this. The `Parser` class has a property called `path`, but this is handling a different `paths` from Swagger.